lms/models/grouping.py

```diff
@@ -23,6 +23,12 @@ class Type(str, Enum):
     __table_args__ = (
         sa.UniqueConstraint("application_instance_id", "authority_provided_id"),
         sa.UniqueConstraint("lms_id", "application_instance_id", "parent_id", "type"),
+        sa.UniqueConstraint("id", "application_instance_id"),
+        sa.ForeignKeyConstraint(
+            ["parent_id", "application_instance_id"],
+            ["grouping.id", "grouping.application_instance_id"],
+            ondelete="cascade",
+        ),
```

Comment:
Moved this `ForeignKey` from the `parent_id` column below into `__table_args__`. When you want a compound foreign key you have to put it in `__table_args__`.

Reply:
Bah, it's just a basic part of how SQLAlchemy works: all multi-column constraints need to be done in the `__table_args__` (things like `ForeignKey(...)` or `unique=True` etc inside a `Column()` are convenience shortcuts that only work for single-column constraints). So I don't think it's worth a comment to explain it as it's just normal SQLAlchemy business